### ISSUE TITLE
Keras neural network regression

### ISSUE THREAD

**Issue by lisamarie95** [rstudio/keras#593]
Hello,

I have a problem with the performance of my neural network. 
**My task:**
I want to predict the annual route of different vehicles. The operation time and the mileage of a vehicle is only known when the vehicle needs to be repaired. (and for some tasks it's important to know this before the car brakes down.) I have a training dataset of vehicles and a test dataset, where I know the annual route of different cars. Additionally I have given several features like the cars type, which fuel it needs and in which country (climate conditions, infra structure..) the vehicle is driving. These (and more) are features the annual driven kilometer depends on. Unfortunately the annual route of all vehicle data I have, is lognormal-distributed. That means for example, that most of the vehicles drive between 10.000 and 20.000 kilometers (normal users), but there are still drivers like cab drivers who reach 40.000 to 60.000 kilometers a year. But in my datasetI don't know who is the driver and if he's gonna drives a lot.

**Image: Lognormal Distribution**
<img width="482" alt="image" src="https://user-images.githubusercontent.com/36514391/48060171-84ec1b00-e1bb-11e8-8c3c-0c9b428f9298.png">

When I now build a neural network, which should predict the annual rout, than I got really bad loss-results.. I think because of the lognormal-distribution the variance is very high, and so with a standard regression neural network it's impossible to get good results.. Am I right? 

**Image Performance**
<img width="533" alt="network_loss" src="https://user-images.githubusercontent.com/36514391/48058027-be218c80-e1b5-11e8-8b5b-cd1cf1633beb.png">

Here is the code of my network..
**Code:**
_______________________________________________________________________________
use_session_with_seed(10)
model <- keras_model_sequential()

model %>%
  layer_dense(units = 400, activation = 'relu',input_shape = ncol(training)) %>% 
  layer_dropout(rate = 0.5) %>%
  layer_dense(units = 400, activation = 'relu') %>%
  layer_dropout(rate = 0.5) %>%
  layer_dense(units = 1)

#compile
model %>% compile(loss = 'mse',
                  optimizer = optimizer_rmsprop(lr = 0.0001),
                  metrics = 'mae')

earlystop <- callback_early_stopping(
  monitor = "val_loss",
  patience = 20)

#fit the model
mymodel <- model %>%
  fit(training,
      trainingtarget,
      epochs = 30,
      batch_size = 20,
      validation_split = 0.30,
      callbacks = list(earlystop),
      shuffle = TRUE,
      verbose = 1)

#evaluate
evalmodel <- model %>% evaluate(test,testtarget)
evalmodel
pred <- model %>% predict(test)
_______________________________________________________________________________

There are other ways to do this? Maybe is it possible to **predict a distribution function of unseen data**? In this case I don't know how to do it. I would be happy about suggestions!!!
Thanks a lot!

**Comment by henry090** [comment]
@lisamarie95 Hi, I think you should normalize your data 
<img width="859" alt="screen shot 2018-11-06 at 16 02 28" src="https://user-images.githubusercontent.com/38523699/48063091-67c84400-e1dd-11e8-95ac-8ace576ac280.png">

**Comment by skeydan** [comment]
How actually are you encoding the training data and how do they look?
(You could e.g. attach a .csv of the first 10 rows).

Also, to see how bad the performance really is, did you compare against other methods (linear regression, SVM, random forest)?
